### PR TITLE
add counter.AddAndLoad

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Introduces a new method to counter: `AddAndLoad(delta) int64`.
AddAndLoad fills a support gap where end users need to handle
synchronicity themselves by locking over an add,value pair of operations.

### CODE REVIEW

File: counter.go [hunk 1]
```diff
@@ -59,21 +59,29 @@ func (c *Counter) Dec() {
...
-		if atomic.CompareAndSwapInt64(&stripe.c, cnt, cnt+delta) {
+		newCnt = cnt + delta
+		if atomic.CompareAndSwapInt64(&stripe.c, cnt, newCnt) {
 			break
 		}
 		// Give a try with another randomly selected stripe.
 		t.idx = runtime_fastrand()
 	}
 	ptokenPool.Put(t)
+	return newCnt
```
Comment (puzpuzpuz) [inline counter.go:84]: This function currently doesn't return the correct value. See `Value` function: it calculates a sum of counter stripes.

Comment (ryanfkeepers) [reply counter.go:84]: Hah, well, that's what I get for only looking at half of the code.  Seems like this won't be useful to pursue.  Thanks for checking it!

Comment (puzpuzpuz) [reply counter.go:84]: No prob at all. The main idea of the counter is that incrementing it from multiple goroutines is much cheaper that with a single atomic int64, but this comes at the cost of more expensive reads.

Comment (puzpuzpuz) [reply counter.go:84]: Thanks for the contribution in any case.

**Pull request closed without merging**